src/main/java/com/abdullah/coding/challenge/repository/CabRepository.java

```diff
@@ -0,0 +1,11 @@
+package com.abdullah.coding.challenge.repository;
+
+import com.abdullah.coding.challenge.entities.Cab;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface CabRepository extends JpaRepository<Cab, Integer> {
+    @Query(value = "SELECT * FROM cab WHERE vehicle_type = :vehicleType AND status = :statusType AND blocked = 0 LIMIT 1", nativeQuery = true)
```

Comment:
I found it more appropriate to use a native query because that is more readable in this instance than writing a routine JPA with their specific format.